jobbergate-agent/tests/jobbergate/test_submit.py

```diff
@@ -519,6 +520,52 @@ async def test_submit_job_script__raises_exception_if_no_executable_script_was_f
     )
 
 
+@pytest.mark.usefixtures("mock_access_token")
+async def test_submit_job_script__raises_exception_if_execution_dir_does_not_exist(
+    dummy_pending_job_submission_data, mocker, user_mapper
+):
+    pending_job_submission = PendingJobSubmission(**dummy_pending_job_submission_data)
+    pending_job_submission.execution_directory = Path("/non/existing/path")
+
+    mock_mark_as_rejected = mocker.patch("jobbergate_agent.jobbergate.submit.mark_as_rejected")
+
+    mocked_sbatch = mock.MagicMock()
+    mocker.patch("jobbergate_agent.jobbergate.submit.SubmissionHandler", return_value=mocked_sbatch)
+
+    with pytest.raises(JobSubmissionError, match="The submission directory must exist and be an absolute path"):
+        await submit_job_script(pending_job_submission, user_mapper)
+
+    mock_mark_as_rejected.assert_called_once_with(
+        dummy_pending_job_submission_data["id"],
+        RegexArgMatcher(".*The submission directory must exist and be an absolute path.*"),
+    )
+
+
+@pytest.mark.usefixtures("mock_access_token")
+async def test_submit_job_script__raises_exception_if_execution_dir_is_relative(
+    dummy_pending_job_submission_data, mocker, user_mapper, tmp_path
+):
+    pending_job_submission = PendingJobSubmission(**dummy_pending_job_submission_data)
+    pending_job_submission.execution_directory = Path("./existing/")
+
+    submit_dir = tmp_path / pending_job_submission.execution_directory
+    submit_dir.mkdir()
+    assert submit_dir.is_dir()
```

Comment:
I don't think this assert is necessary. I trust pathlib to make directories when we call `mkdir()`

Reply:
solved in [6ba95c0](https://github.com/omnivector-solutions/jobbergate/pull/504/commits/6ba95c0ab4b60151f2eb70c9fa71b97f3a466a07)